piksi_tools/console/update_view.py

```diff
@@ -506,15 +508,15 @@ def _update_full_firmware_fired(self):
             pass
 
         self._firmware_update_thread = Thread(
-            target=self.manage_firmware_updates, args=("ALL", ))
+            target=self.manage_firmware_updates, args=("ALL",))
         self._firmware_update_thread.start()
 
     def _download_firmware(self):
         """ Download latest firmware from swiftnav.com. """
         self._write('')
 
         # Check that we received the index file from the website.
-        if self.update_dl is None:
+        if self.update_dl is None or self.update_dl.index is None:
```

Comment:
If we don't have an index, I skip this whole thing rather than waiting for an attribute error.